src/x509.cc

```diff
@@ -277,12 +277,16 @@ Local<Value> try_parse(const std::string& dataString) {
     BUF_MEM *bptr;
     BIO_get_mem_ptr(ext_bio, &bptr);
     BIO_set_close(ext_bio, BIO_CLOSE);
-
     char *data = (char*) malloc(bptr->length + 1);
-    BUF_strlcpy(data, bptr->data, bptr->length + 1);
-    data = trim(data, bptr->length);
 
-    BIO_free(ext_bio);
+    if (bptr->data == NULL){
```

Comment:
Seems like the better option would be to not run the allocation or anything that deals with data unless bptr->data != NULL.